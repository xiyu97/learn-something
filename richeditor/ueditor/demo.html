<!DOCTYPE HTML>
<html>

<head>
    <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
    <title></title>
    <script type="text/javascript" charset="utf-8" src="./ueditor.config.js"></script>
    <script>
        (function () {
            var paths = [
                'editor.js',
                'core/browser.js',
                'core/utils.js',
                'core/EventBase.js',
                'core/dtd.js',
                'core/domUtils.js',
                'core/Range.js',
                'core/Selection.js',
                'core/Editor.js',
                'core/Editor.defaultoptions.js',
                'core/loadconfig.js',
                'core/ajax.js',
                'core/filterword.js',
                'core/node.js',
                'core/htmlparser.js',
                'core/filternode.js',
                'core/plugin.js',
                'core/keymap.js',
                'core/localstorage.js',
                'plugins/defaultfilter.js',
                'plugins/inserthtml.js',
                'plugins/autotypeset.js',
                'plugins/autosubmit.js',
                'plugins/background.js',
                'plugins/image.js',
                'plugins/justify.js',
                'plugins/font.js',
                'plugins/link.js',
                'plugins/iframe.js',
                'plugins/scrawl.js',
                'plugins/removeformat.js',
                'plugins/blockquote.js',
                'plugins/convertcase.js',
                'plugins/indent.js',
                'plugins/print.js',
                'plugins/preview.js',
                'plugins/selectall.js',
                'plugins/paragraph.js',
                'plugins/directionality.js',
                'plugins/horizontal.js',
                'plugins/time.js',
                'plugins/rowspacing.js',
                'plugins/lineheight.js',
                'plugins/insertcode.js',
                'plugins/cleardoc.js',
                'plugins/anchor.js',
                'plugins/wordcount.js',
                'plugins/pagebreak.js',
                'plugins/wordimage.js',
                'plugins/dragdrop.js',
                'plugins/undo.js',
                'plugins/copy.js',
                'plugins/paste.js',
                'plugins/puretxtpaste.js',
                'plugins/list.js',
                'plugins/source.js',
                'plugins/enterkey.js',
                'plugins/keystrokes.js',
                'plugins/fiximgclick.js',
                'plugins/autolink.js',
                'plugins/autoheight.js',
                'plugins/autofloat.js',
                'plugins/video.js',
                'plugins/table.core.js',
                'plugins/table.cmds.js',
                'plugins/table.action.js',
                'plugins/table.sort.js',
                'plugins/contextmenu.js',
                'plugins/shortcutmenu.js',
                'plugins/basestyle.js',
                'plugins/elementpath.js',
                'plugins/formatmatch.js',
                'plugins/searchreplace.js',
                'plugins/customstyle.js',
                'plugins/catchremoteimage.js',
                'plugins/snapscreen.js',
                'plugins/insertparagraph.js',
                'plugins/webapp.js',
                'plugins/template.js',
                'plugins/music.js',
                'plugins/autoupload.js',
                'plugins/autosave.js',
                'plugins/charts.js',
                'plugins/section.js',
                'plugins/simpleupload.js',
                'plugins/serverparam.js',
                'plugins/insertfile.js',
                'ui/ui.js',
                'ui/uiutils.js',
                'ui/uibase.js',
                'ui/separator.js',
                'ui/mask.js',
                'ui/popup.js',
                'ui/colorpicker.js',
                'ui/tablepicker.js',
                'ui/stateful.js',
                'ui/button.js',
                'ui/splitbutton.js',
                'ui/colorbutton.js',
                'ui/tablebutton.js',
                'ui/autotypesetpicker.js',
                'ui/autotypesetbutton.js',
                'ui/cellalignpicker.js',
                'ui/pastepicker.js',
                'ui/toolbar.js',
                'ui/menu.js',
                'ui/combox.js',
                'ui/dialog.js',
                'ui/menubutton.js',
                'ui/multiMenu.js',
                'ui/shortcutmenu.js',
                'ui/breakline.js',
                'ui/message.js',
                'adapter/editorui.js',
                'adapter/editor.js',
                'adapter/message.js',
                'adapter/autosave.js'

            ],
            baseURL = './_src/';
            for (var i = 0, pi;pi = paths[i++];) {
                console.log(i, pi)
                document.write('<script type="text/javascript" src="' + baseURL + pi + '"><\/script>');
            }
        })();
    </script>
    <!--建议手动加在语言，避免在ie下有时因为加载语言失败导致编辑器加载失败-->
    <!--这里加载的语言文件会覆盖你在配置项目里添加的语言类型，比如你在配置项目里配置的是英文，这里加载的中文，那最后就是中文-->
    <script type="text/javascript" charset="utf-8" src="./lang/zh-cn/zh-cn.js"></script>
</head>

<body>
    <!--style给定宽度可以影响编辑器的最终宽度-->
    <script type="text/plain" id="myEditor">
        <p>这里可以写一些输入提示</p>
    </script>
    <script type="text/javascript">
        var editor = UE.getEditor('myEditor', {
            //这里可以选择自己需要的工具按钮名称,此处仅选择如下五个
            toolbars: [[
                // 'anchor', //锚点
                'undo', //撤销
                'redo', //重做
                'bold', //加粗
                'indent', //首行缩进
                // 'snapscreen', //截图
                'italic', //斜体
                'underline', //下划线
                // 'strikethrough', //删除线
                // 'subscript', //下标
                // 'fontborder', //字符边框
                // 'superscript', //上标
                // 'formatmatch', //格式刷
                // 'source', //源代码
                // 'blockquote', //引用
                'pasteplain', //纯文本粘贴模式
                'selectall', //全选
                // 'print', //打印
                // 'preview', //预览
                'horizontal', //分隔线
                'removeformat', //清除格式
                // 'time', //时间
                // 'date', //日期
                // 'unlink', //取消链接
                // 'insertrow', //前插入行
                // 'insertcol', //前插入列
                // 'mergeright', //右合并单元格
                // 'mergedown', //下合并单元格
                // 'deleterow', //删除行
                // 'deletecol', //删除列
                // 'splittorows', //拆分成行
                // 'splittocols', //拆分成列
                // 'splittocells', //完全拆分单元格
                // 'deletecaption', //删除表格标题
                // 'inserttitle', //插入标题
                // 'mergecells', //合并多个单元格
                // 'deletetable', //删除表格
                // 'cleardoc', //清空文档
                // 'insertparagraphbeforetable', //"表格前插入行"
                'insertcode', //代码语言
                'fontfamily', //字体
                'fontsize', //字号
                // 'paragraph', //段落格式
                'simpleupload', //单图上传
                'insertimage', //多图上传
                // 'edittable', //表格属性
                // 'edittd', //单元格属性
                // 'link', //超链接
                'emotion', //表情
                'spechars', //特殊字符
                // 'searchreplace', //查询替换
                // 'map', //Baidu地图
                // 'gmap', //Google地图
                // 'insertvideo', //视频
                // 'help', //帮助
                'justifyleft', //居左对齐
                'justifyright', //居右对齐
                'justifycenter', //居中对齐
                'justifyjustify', //两端对齐
                'forecolor', //字体颜色
                'backcolor', //背景色
                // 'insertorderedlist', //有序列表
                // 'insertunorderedlist', //无序列表
                // 'fullscreen', //全屏
                // 'directionalityltr', //从左向右输入
                // 'directionalityrtl', //从右向左输入
                // 'rowspacingtop', //段前距
                // 'rowspacingbottom', //段后距
                // 'pagebreak', //分页
                // 'insertframe', //插入Iframe
                // 'imagenone', //默认
                // 'imageleft', //左浮动
                // 'imageright', //右浮动
                'attachment', //附件
                // 'imagecenter', //居中
                // 'wordimage', //图片转存
                'lineheight', //行间距
                // 'edittip ', //编辑提示
                'customstyle', //自定义标题
                // 'autotypeset', //自动排版
                // 'webapp', //百度应用
                // 'touppercase', //字母大写
                // 'tolowercase', //字母小写
                // 'background', //背景
                // 'template', //模板
                // 'scrawl', //涂鸦
                // 'music', //音乐
                // 'inserttable', //插入表格
                // 'drafts', // 从草稿箱加载
                // 'charts', // 图表
            ]],
            //focus时自动清空初始化时的内容
            autoClearinitialContent: true,
            //关闭字数统计
            wordCount: false,
            //关闭elementPath
            elementPathEnabled: false,
            //默认的编辑区域高度
            initialFrameHeight: 300
            //更多其他参数，请参考ueditor.config.js中的配置项
        })
    </script>

</body>


</html>
