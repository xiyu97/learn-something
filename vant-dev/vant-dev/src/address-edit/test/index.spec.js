import { AddressEdit } from '..';
import { areaList } from '../../area/demo/area-simple';
import { mount, later, trigger } from '../../../test';

const defaultAddressInfo = {
  name: '测试',
  tel: '13000000000',
  province: '北京市',
  city: '北京市',
  county: '朝阳区',
  addressDetail: 'address detail',
  areaCode: '110101',
  postalCode: '10000',
  isDefault: true,
};

const createComponent = (addressInfo = {}) => {
  const wrapper = mount(AddressEdit, {
    props: {
      areaList,
      addressInfo: {
        ...defaultAddressInfo,
        ...addressInfo,
      },
      showPostal: true,
      showSetDefault: true,
    },
  });

  const button = wrapper.find('.van-button');
  const fields = wrapper.findAll('.van-field');
  return {
    vm: wrapper.vm,
    fields,
    button,
    wrapper,
  };
};

test('should render AddressEdit correctly', () => {
  expect(mount(AddressEdit).html()).toMatchSnapshot();
});

test('should render AddressEdit with props correctly', () => {
  const wrapper = mount(AddressEdit, {
    props: {
      areaList,
      addressInfo: defaultAddressInfo,
      showPostal: true,
      showSetDefault: true,
      showSearchResult: true,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

// test('set-default', () => {
//   const { wrapper } = createComponent();
//   wrapper.find('.van-switch').trigger('click');

//   expect(wrapper.html()).toMatchSnapshot();
// });

test('should allow to custom validator with validator prop', async () => {
  const wrapper = mount(AddressEdit, {
    props: {
      areaList,
      validator: (key, value) => `foo ${key}${value}`,
    },
  });

  const button = wrapper.find('.van-button');
  await button.trigger('click');
  expect(wrapper.find('.van-field__error-message').html()).toMatchSnapshot();
});

test('should valid name and render error message correctly', async () => {
  const { fields, button } = createComponent({
    name: '',
  });

  await button.trigger('click');
  expect(fields[0].html()).toMatchSnapshot();
  await fields[0].find('input').trigger('focus');
  expect(fields[0].html()).toMatchSnapshot();
});

test('should valid tel and render error message correctly', async () => {
  const { fields, button } = createComponent({
    tel: '',
  });

  await button.trigger('click');
  expect(fields[1].html()).toMatchSnapshot();
  await fields[1].find('input').trigger('focus');
  expect(fields[1].html()).toMatchSnapshot();
});

test('should valid area code and render error message correctly', async () => {
  const { fields, button } = createComponent({
    areaCode: '',
  });

  await button.trigger('click');
  expect(fields[2].html()).toMatchSnapshot();
  await fields[2].find('input').trigger('focus');
  expect(fields[2].html()).toMatchSnapshot();
});

test('should valid address detail and render error message correctly', async () => {
  const { fields, button } = createComponent({
    addressDetail: '',
  });

  await button.trigger('click');
  expect(fields[3].html()).toMatchSnapshot();
  await fields[3].find('textarea').trigger('focus');
  expect(fields[3].html()).toMatchSnapshot();
});

test('should valid postal code and render error message correctly', async () => {
  const { fields, button } = createComponent({
    postalCode: '123',
  });

  await button.trigger('click');
  expect(fields[4].html()).toMatchSnapshot();
  await fields[4].find('input').trigger('focus');
  expect(fields[4].html()).toMatchSnapshot();
});

test('should emit change-detail event after changing address detail', () => {
  const wrapper = mount(AddressEdit);
  const field = wrapper.findAll('.van-field__control')[3];

  field.element.value = '123';
  field.trigger('input');
  expect(wrapper.emitted('change-detail')[0][0]).toEqual('123');
});

test('should return current areas after calling getArea method', () => {
  const wrapper = mount(AddressEdit, {
    props: { areaList },
  });

  expect(wrapper.vm.getArea()).toEqual([
    { code: '110000', name: '北京市' },
    { code: '110100', name: '北京市' },
    { code: '110101', name: '东城区' },
  ]);
});

test('should update current areas after calling setAreaCode method', async () => {
  const wrapper = mount(AddressEdit, {
    props: { areaList },
  });

  wrapper.vm.setAreaCode('110102');
  await later();
  expect(wrapper.vm.getArea()).toEqual([
    { code: '110000', name: '北京市' },
    { code: '110100', name: '北京市' },
    { code: '110102', name: '西城区' },
  ]);
});

test('should show search result after focusing to address detail', async () => {
  const wrapper = mount(AddressEdit, {
    props: {
      showSearchResult: true,
      searchResult: [
        { name: 'name1', address: 'address1' },
        { name: 'name2' },
        { address: 'address2' },
      ],
    },
  });

  const field = wrapper.findAll('.van-field__control')[3];
  const input = field.element;
  await field.trigger('focus');

  const items = wrapper.findAll('.van-icon-location-o');
  items[0].element.parentNode.click();
  await later();
  expect(input.value).toEqual('address1 name1');

  items[1].element.parentNode.click();
  await later();
  expect(input.value).toEqual('name2');

  items[2].element.parentNode.click();
  await later();
  expect(input.value).toEqual('address2');

  await field.trigger('blur');
  await later(150);
  expect(wrapper.vm.detailFocused).toBeFalsy();
});

test('should emit delete event after clicking the delete button', async () => {
  const wrapper = mount(AddressEdit, {
    props: {
      showDelete: true,
    },
  });

  const deleteButton = wrapper.findAll('.van-button')[1];
  deleteButton.trigger('click');
  await later();
  document.querySelector('.van-dialog__confirm').click();
  await later();
  expect(wrapper.emitted('delete')).toBeTruthy();
});

test('should emit cancel-delete event after canceling deletion', async () => {
  const wrapper = mount(AddressEdit, {
    props: {
      showDelete: true,
    },
  });

  const deleteButton = wrapper.findAll('.van-button')[1];
  deleteButton.trigger('click');
  await later();
  document.querySelector('.van-dialog__cancel').click();
  await later();
  expect(wrapper.emitted('cancel-delete')).toBeTruthy();
});

test('should update address detail after calling the setAddressDetail method', async () => {
  const { vm, wrapper } = createComponent();
  const textarea = wrapper.find('.van-address-edit-detail').find('textarea');

  expect(textarea.element.value).toEqual('address detail');

  vm.setAddressDetail('test');
  await later();
  expect(textarea.element.value).toEqual('test');
});

test('should emit click-area event after clicking the area field', () => {
  const wrapper = mount(AddressEdit, {
    props: {
      disableArea: true,
    },
  });

  const field = wrapper.findAll('.van-field')[2];
  field.trigger('click');
  expect(wrapper.emitted('click-area')[0]).toBeTruthy();
});

test('should limit tel maxlength when using tel-maxlength prop', () => {
  const wrapper = mount(AddressEdit, {
    props: {
      telMaxlength: 4,
    },
  });

  const telInput = wrapper.find('input[type="tel"]');
  telInput.element.value = '123456';
  trigger(telInput, 'input');

  expect(telInput.element.value).toEqual('1234');
});
