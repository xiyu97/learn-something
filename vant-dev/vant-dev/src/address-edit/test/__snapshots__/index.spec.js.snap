// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow to custom validator with validator prop 1`] = `
<div class="van-field__error-message">
  foo name
</div>
`;

exports[`should render AddressEdit correctly 1`] = `
<div class="van-address-edit">
  <div class="van-address-edit__fields">
    <div class="van-cell van-field">
      <div class="van-cell__title van-field__label">
        <span>
          Name
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="text"
                 class="van-field__control"
                 placeholder="Name"
          >
        </div>
      </div>
    </div>
    <div class="van-cell van-field">
      <div class="van-cell__title van-field__label">
        <span>
          Phone
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="tel"
                 class="van-field__control"
                 placeholder="Phone"
          >
        </div>
      </div>
    </div>
    <div class="van-cell van-cell--clickable van-field"
         role="button"
         tabindex="0"
    >
      <div class="van-cell__title van-field__label">
        <span>
          Area
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="text"
                 class="van-field__control"
                 readonly
                 placeholder="Area"
          >
        </div>
      </div>
      <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
      </i>
    </div>
    <div class="van-cell van-field van-address-edit-detail">
      <div class="van-cell__title van-field__label">
        <span>
          Address
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <textarea rows="1"
                    class="van-field__control"
                    placeholder="Address"
          >
          </textarea>
        </div>
      </div>
    </div>
  </div>
  <div class="van-address-edit__buttons">
    <button type="button"
            class="van-button van-button--danger van-button--normal van-button--block van-button--round van-address-edit__button"
    >
      <div class="van-button__content">
        <span class="van-button__text">
          Save
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`should render AddressEdit with props correctly 1`] = `
<div class="van-address-edit">
  <div class="van-address-edit__fields">
    <div class="van-cell van-field">
      <div class="van-cell__title van-field__label">
        <span>
          Name
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="text"
                 class="van-field__control"
                 placeholder="Name"
          >
        </div>
      </div>
    </div>
    <div class="van-cell van-field">
      <div class="van-cell__title van-field__label">
        <span>
          Phone
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="tel"
                 class="van-field__control"
                 placeholder="Phone"
          >
        </div>
      </div>
    </div>
    <div class="van-cell van-cell--clickable van-field"
         role="button"
         tabindex="0"
    >
      <div class="van-cell__title van-field__label">
        <span>
          Area
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="text"
                 class="van-field__control"
                 readonly
                 placeholder="Area"
          >
        </div>
      </div>
      <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
      </i>
    </div>
    <div class="van-cell van-field van-address-edit-detail">
      <div class="van-cell__title van-field__label">
        <span>
          Address
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <textarea rows="1"
                    class="van-field__control"
                    placeholder="Address"
          >
          </textarea>
        </div>
      </div>
    </div>
    <div class="van-cell van-field">
      <div class="van-cell__title van-field__label">
        <span>
          Postal
        </span>
      </div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body">
          <input type="tel"
                 class="van-field__control"
                 placeholder="Postal"
          >
        </div>
      </div>
    </div>
  </div>
  <div class="van-cell van-cell--center van-address-edit__default">
    <div class="van-cell__title">
      <span>
        Set as the default address
      </span>
    </div>
    <div role="switch"
         class="van-switch van-switch--on"
         style="font-size: 24px;"
         aria-checked="true"
    >
      <div class="van-switch__node">
      </div>
    </div>
  </div>
  <div class="van-address-edit__buttons">
    <button type="button"
            class="van-button van-button--danger van-button--normal van-button--block van-button--round van-address-edit__button"
    >
      <div class="van-button__content">
        <span class="van-button__text">
          Save
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`should valid address detail and render error message correctly 1`] = `
<div class="van-cell van-field van-address-edit-detail">
  <div class="van-cell__title van-field__label">
    <span>
      Address
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <textarea rows="1"
                class="van-field__control"
                placeholder="Address"
                style="height: auto;"
      >
      </textarea>
    </div>
    <div class="van-field__error-message">
      Address can not be empty
    </div>
  </div>
</div>
`;

exports[`should valid address detail and render error message correctly 2`] = `
<div class="van-cell van-field van-address-edit-detail">
  <div class="van-cell__title van-field__label">
    <span>
      Address
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <textarea rows="1"
                class="van-field__control"
                placeholder="Address"
                style="height: auto;"
      >
      </textarea>
    </div>
  </div>
</div>
`;

exports[`should valid area code and render error message correctly 1`] = `
<div class="van-cell van-cell--clickable van-field"
     role="button"
     tabindex="0"
>
  <div class="van-cell__title van-field__label">
    <span>
      Area
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="text"
             class="van-field__control"
             readonly
             placeholder="Area"
      >
    </div>
    <div class="van-field__error-message">
      Please select a receiving area
    </div>
  </div>
  <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
  </i>
</div>
`;

exports[`should valid area code and render error message correctly 2`] = `
<div class="van-cell van-cell--clickable van-field"
     role="button"
     tabindex="0"
>
  <div class="van-cell__title van-field__label">
    <span>
      Area
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="text"
             class="van-field__control"
             readonly
             placeholder="Area"
      >
    </div>
  </div>
  <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
  </i>
</div>
`;

exports[`should valid name and render error message correctly 1`] = `
<div class="van-cell van-field">
  <div class="van-cell__title van-field__label">
    <span>
      Name
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="text"
             class="van-field__control"
             placeholder="Name"
      >
    </div>
    <div class="van-field__error-message">
      Please fill in the name
    </div>
  </div>
</div>
`;

exports[`should valid name and render error message correctly 2`] = `
<div class="van-cell van-field">
  <div class="van-cell__title van-field__label">
    <span>
      Name
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="text"
             class="van-field__control"
             placeholder="Name"
      >
    </div>
  </div>
</div>
`;

exports[`should valid postal code and render error message correctly 1`] = `
<div class="van-cell van-field">
  <div class="van-cell__title van-field__label">
    <span>
      Postal
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="tel"
             class="van-field__control"
             placeholder="Postal"
      >
    </div>
    <div class="van-field__error-message">
      Wrong postal code
    </div>
  </div>
</div>
`;

exports[`should valid postal code and render error message correctly 2`] = `
<div class="van-cell van-field">
  <div class="van-cell__title van-field__label">
    <span>
      Postal
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="tel"
             class="van-field__control"
             placeholder="Postal"
      >
    </div>
  </div>
</div>
`;

exports[`should valid tel and render error message correctly 1`] = `
<div class="van-cell van-field">
  <div class="van-cell__title van-field__label">
    <span>
      Phone
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="tel"
             class="van-field__control"
             placeholder="Phone"
      >
    </div>
    <div class="van-field__error-message">
      Malformed phone number
    </div>
  </div>
</div>
`;

exports[`should valid tel and render error message correctly 2`] = `
<div class="van-cell van-field">
  <div class="van-cell__title van-field__label">
    <span>
      Phone
    </span>
  </div>
  <div class="van-cell__value van-field__value">
    <div class="van-field__body">
      <input type="tel"
             class="van-field__control"
             placeholder="Phone"
      >
    </div>
  </div>
</div>
`;
